{{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/tests/bundles/test_example_bundle.py

```diff
@@ -1,53 +1,75 @@
-import tempfile
-from pathlib import Path
-
 from hamcrest import assert_that, contains, has_properties
-from microcosm_sagemaker.artifact import InputArtifact, OutputArtifact
-from microcosm_sagemaker.input_data import InputData
-from microcosm_sagemaker.testing.directory_comparison import (
-    directory_comparison,
+from microcosm_sagemaker.testing.bundle import (
+    BundleFitTestCase,
+    BundleLoadTestCase,
+    BundleSaveTestCase,
+    BundleTestCase,
 )
 
-from {{ cookiecutter.package_name }}.app_hooks.train.app import create_app
+from {{ cookiecutter.package_name }}.bundles.example import ExampleBundle
 from {{ cookiecutter.package_name }}.tests.fixtures import get_fixture_path
 
 
-class TestExampleBundle:
-    def setup(self) -> None:
-        self.graph = create_app(extra_deps=["example_bundle"])
-        self.training_initializers = self.graph.training_initializers
+class ExampleBundleTestCase(BundleTestCase):
+    """
+    All of the test cases in this file will be testing example bundle, and use
+    the same root input artifact.  They will also check the bundle prediction
+    in the same way when necessary.  Thus, we define these parameters in a base
+    class here.
+
+    """
 
-        self.input_data = InputData(get_fixture_path("example_input_data"))
-        self.input_artifact = InputArtifact(get_fixture_path("example_input_artifact"))
-        self.gold_output_artifact_path = get_fixture_path("example_gold_output_artifact")
+    # Determines which bundle will be tested
+    bundle_name = "example_bundle"
 
-    def check_bundle_prediction(self) -> None:
+    # Contains a full root input artifact that will be used to automatically
+    # load all dependency bundles and, for the load test, to test loading of
+    # the bundle under test
+    root_input_artifact_path = get_fixture_path("example_input_artifact")
+
+    # This function will be called on the bundle after fitting / loading to
+    # make sure it works correctly
+    def check_bundle_prediction(self, bundle: ExampleBundle) -> None:
         assert_that(
-            self.graph.example_bundle.predict(1.0),
+            bundle.predict(1.0),
             contains(has_properties(
                 uri="http://example.com",
                 score=3.0,
             )),
         )
 
-    def test_fit(self) -> None:
-        self.training_initializers.init()
-        self.graph.example_bundle.fit(self.input_data)
-        self.check_bundle_prediction()
 
-    def test_load(self) -> None:
-        self.graph.example_bundle.load(self.input_artifact)
-        self.check_bundle_prediction()
+class TestExampleBundleFit(BundleFitTestCase, ExampleBundleTestCase):
+    """
+    We just need to provide the location of a small test dataset.  This dataset
+    will be used to fit the bundle, and then the prediction will be tested
+    using the `check_bundle_prediction` function defined above.
+
+    """
+    input_data_path = get_fixture_path("example_input_data")
```

Comment:
I thought it would be clearer to move the common parameters to the base class, and define the rest only in the test case where they are needed.  This way it is clear exactly what each parameter is used for.  Make sense?